src/countries/countries.service.ts

```diff
@@ -1,4 +1,46 @@
 import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Country } from './country.entity';
+import { DeleteResult, Repository, UpdateResult } from 'typeorm';
+import { CountryInterface } from './country.interface';
 
 @Injectable()
-export class CountriesService {}
+export class CountriesService {
+  constructor(
+    @InjectRepository(Country)
+    private countryRepository: Repository<CountryInterface>,
+  ) {}
+
+  findAll(): Promise<CountryInterface[]> {
+    return this.countryRepository.find();
+  }
+
+  findOne(id: number): Promise<CountryInterface | null> {
```

Comment:
Might be a personal opinion, but I prefer to have the service doing the promise settlement, and why I think this is the best thing to do is the following example:
```typescript
async findOne(id: number): Promise<CountryDto> {
    const country = await this.countryRepository.findOneBy({ id });
    if (!country) throw new CountryNotFoundError(id);
    return country;
  }
``` 
Allowing the service to fetch and check and throw a custom exception will make the controller much cleaner.